### DISCUSSION TITLE
Which fields are required for ESP32? "hardware is 'ESP32 (excl S2-S3/C2-C6)', restore file 'config.json' is for hardware 'ESP82'"

### DISCUSSION THREAD

**Discussion by ejohb** [tasmota/decode-config#95] [Q&A] [answered]
### Describe the bug

_A clear and concise description of what the bug is._

Not really a bug, but I cannot find the fields necessary to do a partial restore on non-ESP86 devices. I have added the entire `header` structure from a full backup but still something is missing. Is this documented anywhere?

### To Reproduce

Using the following partial config:

```
{
  "devicename": "Infrastructure Energy",
  "header": {
    "data": {
      "crc": "0x45ef",
      "crc32": "0xd66ca3c6",
      "hardware": "ESP32 (excl S2-S3/C2-C6)",
      "size": 4096,
      "version": {
        "id": "0xe030000",
        "name": "14.3.0"
      }
    },
    "env": {
      "platform": "Linux-6.11.0-9-generic-x86_64-with-glibc2.40",
      "python": "3.12.7",
      "script": "decode-config.py v14.3.0.0 [4266903]",
      "system": "Linux x86_64 6.11.0-9-generic #9-Ubuntu SMP PREEMPT_DYNAMIC Mon Oct 14 13:19:59 UTC 2024"
    },
    "template": {
      "crc": "0x8d12",
      "crc32": "0xd66ca3c6",
      "size": 4096,
      "version": {
        "id": "0xe030000",
        "name": "14.3.0"
      }
    },
    "timestamp": "2024-11-18 10:31:36"
  }
}
```

Results in the following message:

```
decode-config.py --source <address> --restore-file config.json
ERROR 9: Restore data incompatibility: Device '<address>' hardware is 'ESP32 (excl S2-S3/C2-C6)', restore file 'config.json' is for hardware 'ESP82'
Premature exit - #9 Restore data error
```

### Expected behavior

Some documentation on which fields are required for ESP32 etc.

### Version Information

_Please complete the following information_  
_use `decode-config.py -V --debug` to retrieve the info_

```bash
decode-config.py v14.3.0.0 [4266903] by Norbert Richter nr@prsolution.eu

Script:   decode-config.py
Version:  14.3.0.0 [4266903]
Python:   3.12.7
Platform: Linux-6.11.0-9-generic-x86_64-with-glibc2.40 - x86_64
OS:       Linux 6.11.0-9-generic #9-Ubuntu SMP PREEMPT_DYNAMIC Mon Oct 14 13:19:59 UTC 2024
Time:     2024-11-18 11:46:25
```

### System' information of your Tasmota data

_Append `-g System` to your decode-config parameter, e. g. `decode-config.py -s myconfig.dmp -g System`_

N/A

**Comment by curzon01** [comment, marked as answer]
complete `header` is for information only, not used for restore and can be omitted.

`config_version` determines which hardware platform is meant and is mandatory for all non-ESP82xx chip types when using json subsets.

This is documented under _Note_ on [Restore subset of data](https://github.com/tasmota/decode-config?tab=readme-ov-file#restore-subset-of-data)

See also #91 and [answer](https://github.com/tasmota/decode-config/issues/91#issuecomment-2259929327)

**Reply by ejohb** [reply]
Oh yes, I must have missed that. Many thanks for your help!
